### ISSUE TITLE
Hyperparameter

### ISSUE THREAD

**Issue by MHpurity** [scu-zjz/SparseViT#16]
The parameters of train.sh files are different from your paper. Could you tell me which one is the right one?
train.sh :
main_train.py \
    --world_size 1 \
    --batch_size 16 \
    --data_path "<The location of your combined dataset configuration file>" \
    --epochs 200 \
    --lr 2e-4 \
    --min_lr 5e-7 \
    --weight_decay 0.05 \
    --pretrain_path "<path to uniformer_base_ls_in1k.pth>" \
    --test_data_path "<path to /CASIA1.0>" \
    --warmup_epochs 4 \
    --output_dir ./output_dir/ \
    --log_dir ./output_dir/  \
    --accum_iter 1 \
    --seed 42 \
    --test_period 4 \
    --num_workers 8 \
The paper's :
[During training, we selected a batch size of 16 and set 200 training epochs to ensure that the model could fully learn and converge. For optimization, we used the Adam optimizer with an initial learning rate of 1 × 10−4 , which was then periodically decayed to 1 × 10−7 using a cosine annealing strategy.]

**Comment by MHpurity** [comment]
> The parameters of train.sh files are different from your paper. Could you tell me which one is the right one? train.sh : main_train.py --world_size 1 --batch_size 16 --data_path "" --epochs 200 --lr 2e-4 --min_lr 5e-7 --weight_decay 0.05 --pretrain_path "" --test_data_path "<path to /CASIA1.0>" --warmup_epochs 4 --output_dir ./output_dir/ --log_dir ./output_dir/ --accum_iter 1 --seed 42 --test_period 4 --num_workers 8 The paper's : [During training, we selected a batch size of 16 and set 200 training epochs to ensure that the model could fully learn and converge. For optimization, we used the Adam optimizer with an initial learning rate of 1 × 10−4 , which was then periodically decayed to 1 × 10−7 using a cosine annealing strategy.]

Also, the optimizer is AdamW in your codes, is it ok to train the model with the optimizer?

**Comment by DSLJDI** [comment]
Thank you for your attention to our work. I believe the “difference” you mentioned refers to the variation in the initial learning rates. I think the results you obtained with these two different learning rates are not significantly different. For the paper, there’s no need to get caught up in the so-called learning rate difference — both settings work.

**Comment by MHpurity** [comment]
> Thank you for your attention to our work. I believe the “difference” you mentioned refers to the variation in the initial learning rates. I think the results you obtained with these two different learning rates are not significantly different. For the paper, there’s no need to get caught up in the so-called learning rate difference — both settings work.

Thanks for you reply.
I have trained the model using two different learning rates. But both f1-scores are about 6% less than your paper's. Am I missing something, or could you please advise?

**Comment by DSLJDI** [comment]
> > Thank you for your attention to our work. I believe the “difference” you mentioned refers to the variation in the initial learning rates. I think the results you obtained with these two different learning rates are not significantly different. For the paper, there’s no need to get caught up in the so-called learning rate difference — both settings work.
> 
> Thanks for you reply. I have trained the model using two different learning rates. But both f1-scores are about 6% less than your paper's. Am I missing something, or could you please advise?

If your training dataset is exactly the same as ours, it’s possible that differences in GPU hardware may have caused some variation. Could you please check whether you adjusted the batch size during training? If so, you may need to find a learning rate that matches the new batch size.

Additionally, we found that gradient accumulation has a significant impact on SparseViT’s performance, so you might want to experiment with that as well. You can also try adjusting the dropout rate, as it has a considerable influence on the model’s behavior.

**Comment by MHpurity** [comment]
> > > Thank you for your attention to our work. I believe the “difference” you mentioned refers to the variation in the initial learning rates. I think the results you obtained with these two different learning rates are not significantly different. For the paper, there’s no need to get caught up in the so-called learning rate difference — both settings work.
> > 
> > 
> > Thanks for you reply. I have trained the model using two different learning rates. But both f1-scores are about 6% less than your paper's. Am I missing something, or could you please advise?
> 
> If your training dataset is exactly the same as ours, it’s possible that differences in GPU hardware may have caused some variation. Could you please check whether you adjusted the batch size during training? If so, you may need to find a learning rate that matches the new batch size.
> 
> Additionally, we found that gradient accumulation has a significant impact on SparseViT’s performance, so you might want to experiment with that as well. You can also try adjusting the dropout rate, as it has a considerable influence on the model’s behavior.

I uesd one 16GB RTX 40x GPU to train the model . I didn't modify the batchsize and just adjusted the numworker . However, I am not sure if I processed the datasets correctly.
